api/src/main/java/edu/wpi/first/shuffleboard/api/components/IntegerField.java

```diff
@@ -0,0 +1,74 @@
+package edu.wpi.first.shuffleboard.api.components;
+
+import edu.wpi.first.shuffleboard.api.util.PropertyUtils;
+
+import java.util.regex.Pattern;
+
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.scene.control.TextField;
+import javafx.scene.control.TextFormatter;
+
+/**
+ * A type of text field that only accepts valid whole numbers.
+ */
+public class IntegerField extends TextField {
+
+  private final Property<Integer> number = new SimpleObjectProperty<>(this, "number", 0);
+  private static final Pattern numberPattern = Pattern.compile("^[-+]?\\d*$");
+
+  /**
+   * Creates a new number field with no value.
+   */
+  public IntegerField() {
```

Comment:
This is not the correct name or semantics for this class, since port numbers aren't integers, they're non-negative integers between the range of 0 and MAX_SHORT